### PR TITLE
#461 Prover: Add constraints version validation

### PR DESCRIPTION
**Pull request by srinathln7**
This PR implements issue(s) #

* Adds a command in the MakeFile under `zkevm/arithmetization/zkevm.bin` to extract the version from the git branch in the constraints repo and creates a `constraints-versions.txt` file. This approach relies on branch naming conventions as the [linea-constraints](https://github.com/Consensys/linea-constraints) repo does not support tags for every release. Tags with version names for every release are recommended to reliably extract versions from tags instead of branches.
* Adds version validation in the prover to ensure the version from the conflated trace file matches the trace engine version and the ones listed in the `constraints-versions.txt` file.
* Code changes and validation checks were successfully tested manually with correct and incorrect trace files in the prover `dev` mode.

### CODE REVIEW

File: prover/Makefile
Comment (AlexandreBelling) [inline prover/Makefile]: IIUC, BRANCHES contains all the branches containing the target commit. This includes the "right" branch but will also include all the future branches. 

E.g. if we run the same comment in 5 months it will return all versions >= `v0.8.0-rc8.1`

File: prover/run.sh
Comment (AlexandreBelling) [inline prover/run.sh]: Can you soft remove this file and also remove the changes made to zkevm.bin

File: prover/config/config.go
Comment (AlexandreBelling) [inline prover/config/config.go]: to remove

File: prover/config/config-sepolia-full.toml
Comment (AlexandreBelling) [inline prover/config/config-sepolia-full.toml]: Soft remove these changes

File: prover/config/config-integration-full.toml
Comment (AlexandreBelling) [inline prover/config/config-integration-full.toml]: Soft remove these changes

File: prover/constraints-versions.txt
Comment (gusiri) [inline prover/constraints-versions.txt]: In the future, it would be great if `constraints-versions.txt` could be in `.json` or another format and include useful information, such as the commit of the constraint, etc.

File: prover/Makefile
Comment (srinathln7) [reply prover/Makefile]: Switched to tags now :)

File: prover/run.sh
Comment (srinathln7) [reply prover/run.sh]: Done :)

File: prover/config/config.go
Comment (srinathln7) [reply prover/config/config.go]: done

File: prover/constraints-versions.txt
Comment (srinathln7) [reply prover/constraints-versions.txt]: noted!

File: prover/config/config-sepolia-full.toml
Comment (srinathln7) [reply prover/config/config-sepolia-full.toml]: done :)

File: prover/config/config-integration-full.toml
Comment (srinathln7) [reply prover/config/config-integration-full.toml]: done :)

File: prover/config/config-integration-full.toml
Comment (gusiri) [inline prover/config/config-integration-full.toml]: Let's use a sample path that actually exists in the repo, for example: `../testdata/prover-v2/conflated`

**Comment by gusiri** [comment]
You can change the title of this PR to something like
`Prover: add constraints version validation` before merging it into the main branch.

File: prover/config/config-integration-full.toml
Comment (AlexandreBelling) [reply prover/config/config-integration-full.toml]: Best to leave it to the default value IMO

Comment (gusiri) [reply prover/config/config-integration-full.toml]: I agree.

Comment (srinathln7) [reply prover/config/config-integration-full.toml]: What is the default value? The `main` does not have this dir set. Shall I remove it?

Comment (gusiri) [reply prover/config/config-integration-full.toml]: `/shared/traces/conflated` is the default value

Comment (gusiri) [reply prover/config/config-integration-full.toml]: and yes you can actually remove it as well.

Comment (AlexandreBelling) [reply prover/config/config-integration-full.toml]: Yes, the default values are passed to the config in case the field is missing in the toml file. The values are defined in `config_default.go` the defined default value is the field that @gusiri pointed above.

My recommendation is undo this change and leave the default value

Comment (srinathln7) [reply prover/config/config-integration-full.toml]: Done :)

File: prover/cmd/prover/cmd/prove.go
Comment (gusiri) [inline prover/cmd/prover/cmd/prove.go]: a typo: used to generated -> used to generate

Comment (srinathln7) [reply prover/cmd/prover/cmd/prove.go]: fixed, good catch :)

**Review by gusiri — APPROVED** [review]
Looks good to me.

**Review by AlexandreBelling — DISMISSED** [review]

**Comment by srinathln7** [comment]
UPDATE: @gusiri  @AlexandreBelling  Reworking this PR as e2e tests are failing due to the absence of `constraints-versions.txt` file in the environment. The plan now is to use `go embed` plugin and moving this file into `backend/execution`.

**Review by AlexandreBelling — DISMISSED** [review]

**Review by gusiri — APPROVED** [review]

**Review by AlexandreBelling — APPROVED** [review]

**Pull request merged**
